src/app/clusters/on-off-server/on-off-server.h

```diff
@@ -96,19 +96,20 @@ struct OnOffEffect
     chip::EndpointId mEndpoint;
     OffWithEffectTriggerCommand mOffWithEffectTrigger = nullptr;
     chip::app::Clusters::OnOff::OnOffEffectIdentifier mEffectIdentifier;
-    uint8_t mEffectVariant;
+    chip::app::Clusters::OnOff::OnOffDelayedAllOffEffectVariant mEffectVariant;
     OnOffEffect * nextEffect = nullptr;
 
-    OnOffEffect(
-        chip::EndpointId endpoint, OffWithEffectTriggerCommand offWithEffectTrigger,
-        chip::app::Clusters::OnOff::OnOffEffectIdentifier effectIdentifier = EMBER_ZCL_ON_OFF_EFFECT_IDENTIFIER_DELAYED_ALL_OFF,
-
-        /*
-         * effectVariant's type depends on the effect effectIdentifier so we don't know the type at compile time.
-         * Casting to uint8_t for more flexibility since the type can be OnOffDelayedAllOffEffectVariant or
-         * OnOffDelayedAllOffEffectVariant
-         */
-        uint8_t effectVariant = static_cast<uint8_t>(EMBER_ZCL_ON_OFF_DELAYED_ALL_OFF_EFFECT_VARIANT_FADE_TO_OFF_IN_0P8_SECONDS));
+    OnOffEffect(chip::EndpointId endpoint, OffWithEffectTriggerCommand offWithEffectTrigger,
+                chip::app::Clusters::OnOff::OnOffEffectIdentifier effectIdentifier =
+                    chip::app::Clusters::OnOff::OnOffEffectIdentifier::kDelayedAllOff,
+
+                /*
+                 * effectVariant's type depends on the effect effectIdentifier so we don't know the type at compile time.
```

Comment:
This comment no longer matches the code?  But the comment is right; the code is what needs to change.

Reply:
I don't know how I missed that comment... Sorry about that.